zerver/views/users.py

```diff
@@ -910,6 +915,25 @@ def get_subscription_backend(
     return json_success(request, data=subscription_status)
 
 
+@typed_endpoint
+def get_subscriptions_backend(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    user_id: PathOnly[Json[int]],
+) -> HttpResponse:
+    target_user = access_user_by_id(user_profile, user_id, for_admin=False)
+    streams = get_streams_for_user(
+        target_user,
+        include_all=True,
+    )
+    subscriptions = [
+        stream.id for stream in streams if subscribed_to_stream(target_user, stream.id)
```

Comment:
This does a database query for each row :(

```
def subscribed_to_stream(user_profile: UserProfile, stream_id: int) -> bool:                            
    return Subscription.objects.filter(                                                                 
        user_profile=user_profile,                                                                      
        active=True,                                                                                    
        recipient__type=Recipient.STREAM,                                                               
        recipient__type_id=stream_id,                                                                   
    ).exists()                                                                                          
```

Reply:
Confirmed the tests had query counts of 15 and 8 beforehand, and 7 and 8 after. (The second gets cut short by invalid access)